src/mainClass.js

```diff
@@ -2,17 +2,24 @@ import { getData, getAllCategories } from "./source";
 
 export class AllItems {
 
-  constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
+  constructor(renderCategories, renderLinks) {
+    // categories, search functionality
     this.categories = [];
-    this.onReceivedCategories = searchCategory;
-    this.displayChosenLinks = displayLinksForChosenCategory;
+    this.onReceivedCategories = renderCategories;
+    // links, search functionality and initial display when app starts
     this.links = [];
-    this.onReceivedLinks = searchLinks;
+    this.onReceivedLinks = renderLinks;
+    // localStorage
     this.storageLinks = JSON.parse(localStorage.getItem('links'));
     this.storageCategories = JSON.parse(localStorage.getItem('categories'));
-    this.textFilter;
+    // used for search
+    this.textFilterLinks;
```

Comment:
this.textFilterLinks;
    this.textFilterCategories;
    this.selectedCategory; 
initialise this values to something best ""